core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -195,12 +198,13 @@ public void seek(long pos) throws IOException {
       return;
     }
     Preconditions.checkArgument(pos >= 0, PreconditionMessage.ERR_SEEK_NEGATIVE, pos);
-    Preconditions.checkArgument(pos < mFileLength, PreconditionMessage.ERR_SEEK_PAST_END_OF_FILE,
-        pos);
-
-    seekBlockInStream(pos);
-    checkAndAdvanceBlockInStream();
-    mCurrentBlockInStream.seek(mPos % mBlockSize);
+    Preconditions
+        .checkArgument(pos < mFileLength, PreconditionMessage.ERR_SEEK_PAST_END_OF_FILE, pos);
+    if (mShouldCacheIncompleteBlock) {
```

Comment:
what if `mShouldCacheCurrentBlock` is `false`? do we still call `seekWithCachingIncompleteBlocks` in this case?